cruise-control/src/yaml/endpoints/rebalance.yaml

```diff
@@ -0,0 +1,200 @@
+RebalanceEndpoint:
+  post:
+    operationId: rebalance
+    summary: Rebalance a Kafka cluster based on goals
+    parameters:
+      - name: dryrun
+        in: query
+        description: Whether dry-run the request or not
+        schema:
+          type: boolean
+          default: true
+      - name: goals
+        in: query
+        description: list of goals used to generate proposal
+        schema:
+          type: array
+          items:
+            type: string
+          # default is every goal
+      - name: allow_capacity_estimation
+        in: query
+        description: Whether to allow broker capacity to be estimated
+        schema:
+          type: boolean
+          default: false
+      - name: concurrent_partition_movements_per_broker
+        in: query
+        description: The upper bound of ongoing replica movements going into/out of each broker
+        schema:
+          type: integer
+          format: int32
+          minimum: 1
+      - name: concurrent_intra_broker_partition_movements
+        in: query
+        description: The upper bound of ongoing replica movements between disks within each broker
+        schema:
+          type: integer
+          format: int32
+          minimum: 1
+      - name: concurrent_leader_movements
+        in: query
+        description: The upper bound of ongoing leadership movements
+        schema:
+          type: integer
+          format: int32
+          minimum: 1
+      - name: json
+        in: query
+        description: Whether to return in JSON format or not
+        schema:
+          type: boolean
+          default: false
+      - name: skip_hard_goal_check
+        in: query
+        description: Whether to allow hard goal be skipped in proposal generation
+        schema:
+          type: boolean
+          default: false
+      - name: excluded_topics
+        in: query
+        description: Specify topic whose partition is excluded from replica movement
+        schema:
+          type: string # topics regex
+          default: null
+      - name: use_ready_default_goals
+        in: query
+        description: Whether to only use ready goals to generate proposal
+        schema:
+          type: boolean
+          default: false
+      - name: data_from
+        in: query
+        description: Whether to calculate proposal from available valid partitions or valid windows
+        schema:
+          type: string
+          default: VALID_WINDOWS
+      - name: verbose
+        in: query
+        description: Return detailed state information
+        schema:
+          type: boolean
+          default: false
+      - name: exclude_recently_demoted_brokers
+        in: query
+        description: Whether to allow replicas to be moved to recently demoted broker
```

Comment:
`replicas` -> `leader replicas`, broker demotion is to transfer all the leadership out of broker, so only leader replica matters here.